cudax/include/cuda/experimental/__hierarchy/hierarchy_dimensions.cuh

```diff
@@ -715,6 +744,47 @@ public:
 
     return ::cuda::std::apply(detail::get_level_helper<Level>{}, levels);
   }
+
+  /**
+   * @brief Returns a hierarchy updated to replace meta dimsnions with concrete dimensions
+   *
+   * This function will create a new hierarchy finalized for the passed in device and kernel. Each level in
+   * the hierarchy will be replaced with concrete dimensions if they are meta dimensions
+   * or passed through otherwise.
+   *
+   * @param device
+   * Device to finalize the dimensions for, for example in case of fill_device meta dimensions
+   *
+   * @param kernel
+   * Kernel that the hierarchy is intended for
+   *
+   * @tparam Args
+   * Types of kernel arguments, need to be explicitly specified only for a kernel functor
+   */
+  template <typename... Args, typename Kernel>
+  _CCCL_NODISCARD auto finalize(device_ref device, const Kernel& kernel);
```

Comment:
What is the purpose of this undefined method?

Reply:
Its defined in `hierarchy_draft.cuh` and its the main `finalize`, the other overload here is only for convenience to spell out argument types easier.